sdk/servicebus/service-bus/src/connectionContext.ts

```diff
@@ -130,6 +136,66 @@ type ConnectionContextMethods = Omit<
 > &
   ThisType<ConnectionContextInternalMembers>;
 
+/**
+ * @internal
+ * @hidden
+ * Helper method to call onDetached on the receivers from the connection context upon seeing an error.
+ */
+async function callOnDetachedOnReceivers(
+  connectionContext: ConnectionContext,
+  contextOrConnectionError: Error | ConnectionError | AmqpError | undefined,
+  receiverType: ReceiverType
+) {
+  const detachCalls: Promise<void>[] = [];
+
+  for (const receiverName of Object.keys(connectionContext.messageReceivers)) {
+    const receiver = connectionContext.messageReceivers[receiverName];
+    if (receiver && receiver.receiverType === receiverType) {
```

Comment:
Does this affect session receivers at all? Just want to make sure we're still calling onDetached on all receivers like we used to now that we're checking receiverType as well.

Reply:
`onDetached` was(and is) only called for the streaming and batching receivers.

(Session receivers can only be accessed through `connectionContext.messageSessions`. We never called `onDetached` for sessions, there is no `onDetached` method that is implemented for sessions.)

Previously, `onDetached` was called for "batching" and "streaming" receivers after the refresh connection.
Now, onDetached is called for "batching" before the refresh connection, unlike the streaming receivers.

I left a TODO to investigate further on the "sessions" part, I'll log an issue for it too.

Reply:
There is this related issue #8875 which intends to deal with the work related to disconnects, that issue should take care of the remaining wor for sessions, not logging a new issue.